lib/logstash/codecs/avro.rb

```diff
@@ -131,4 +183,165 @@ def encode(event)
       @on_event.call(event, buffer.string)
     end  
   end
+
+  private
+  def fetch_schema(uri_string)
+    http_connection = uri_string.start_with?('http://')
+    https_connection = uri_string.start_with?('https://')
+
+    if http_connection
+      ssl_config_provided = original_params.keys.select {|k| k.start_with?("ssl_") && k != "ssl_enabled" }
+      if ssl_config_provided.any?
+        raise_config_error! "When SSL is disabled, the following provided parameters are not allowed: #{ssl_config_provided}"
+      end
+
+      credentials_configured = @username && @password
+      @logger.warn("Credentials are being sent over unencrypted HTTP. This may bring security risk.") if credentials_configured && http_connection
+      fetch_remote_schema(uri_string)
+    elsif https_connection
+      validate_ssl_settings!
+      fetch_remote_schema(uri_string)
+    else
+      # local schema
+      URI.open(uri_string, &:read)
+    end
+  end
+
+  def fetch_remote_schema(uri_string)
+    client_options = {}
+
+    if @proxy && !@proxy.empty?
+      client_options[:proxy] = @proxy.to_s
+    end
+
+    basic_auth_options = build_basic_auth
+    client_options[:auth] = basic_auth_options unless basic_auth_options.empty?
+
+    if @ssl_enabled
+      ssl_options = build_ssl_options
+      client_options[:ssl] = ssl_options unless ssl_options.empty?
+    end
+
+    client = Manticore::Client.new(client_options)
+    response = client.get(uri_string).call
+
+    unless response.code == 200
+      raise "HTTP request failed: #{response.code} #{response.message}"
+    end
+
+    body = response.body
+    # Response may contain schema metadata, schema field is what we need
+    # Example response: {"subject":"test-no-auth-1763597024","version":1,"id":1,"guid":"5c6c5f26-e876-e5ab-02b0-8d9bebbc90d7","schemaType":"AVRO","schema":"{"type":"record","name":"TestRecord","namespace":"com.example","fields":[{"name":"message","type":"string"},{"name":"timestamp","type":"long"}]}","ts":1763597024561,"deleted":false}
+    begin
+      parsed = JSON.parse(body)
+      if parsed.is_a?(Hash) && parsed.has_key?('schema')
+        parsed['schema']
+      else
+        # fallback to use the response as it is
+        body
+      end
+    rescue JSON::ParserError
+      # Not JSON, return as-is (probably a direct schema)
+      body
+    end
```

Comment:
we can unwrap begin end
```suggestion
    parsed = JSON.parse(body)
    if parsed.is_a?(Hash) && parsed.has_key?('schema')
      parsed['schema']
    else
      # fallback to use the response as it is
      body
    end
  rescue JSON::ParserError
    # Not JSON, return as-is (probably a direct schema)
    body
```

Reply:
Applied after confirmation (local tests) with `client = nil`.